src/dstack/_internal/core/backends/gcp/compute.py

```diff
@@ -116,7 +136,68 @@ def create_instance(
             project_id=self.config.project_id,
         )
         disk_size = round(instance_offer.instance.resources.disk.size_mib / 1024)
+        commands = get_shim_commands(authorized_keys=authorized_keys)
+        startup_script = " ".join([" && ".join(commands)])
+        startup_script = "#! /bin/bash\n" + startup_script
+        instance_id = f"tpu-{instance_config.instance_name}"
+        if "tpu" in instance_offer.instance.resources.gpus[0].name:
+            tpu_node = gcp_resources.create_tpu_node_struct(
```

Comment:
create_tpu_node_struct() name is misleading since it not only creates a struct but also does a network call to create a persistent disk. So it may raise GCP errors (which are not handled). I suggest creating persisting disk outside of create_tpu_node_struct() and passing the disk reference there.